add v0.1.7 release notes (#273)
streamnative/function-mesh: CHANGELOG.md

# Function Mesh v0.1.7 Release Notes

## v0.1.7 What's New

> - [mesh-worker-service] add sink/source typeClassName in connector definition ([#247](https://github.com/streamnative/function-mesh/pull/247))

Reviewer: ```suggestion
- [Function Mesh Worker service] Add `typeClassName` in the source and sink connector definition ([#247](https://github.com/streamnative/function-mesh/pull/247))
```

> - [tests] add integration tests for mesh-worker-service ([#246](https://github.com/streamnative/function-mesh/pull/246))

Reviewer: ```suggestion
- [Test] Add integration tests for Function Mesh Worker service ([#246](https://github.com/streamnative/function-mesh/pull/246))
```

> - [controller] use untyped yaml configuration as function or connector config ([#233](https://github.com/streamnative/function-mesh/pull/233))

Reviewer: ```suggestion
- [Controller] Use untyped YAML configuration as function or connector configuration ([#233](https://github.com/streamnative/function-mesh/pull/233))
```

> - [controller] enable source connector to set ForwardMessageProperty ([#250](https://github.com/streamnative/function-mesh/pull/250))

Reviewer: ```suggestion
- [Controller] Enable source connector to set `ForwardMessageProperty` ([#250](https://github.com/streamnative/function-mesh/pull/250))
```

> - [runner-image] fix build python runner image based on the latest pulsar image ([#253](https://github.com/streamnative/function-mesh/pull/253)) ([#254](https://github.com/streamnative/function-mesh/pull/254))

Reviewer: ```suggestion
- [Runner image] Find the issue that the Python runner image cannot be built based on the latest Pulsar image ([#253](https://github.com/streamnative/function-mesh/pull/253)) ([#254](https://github.com/streamnative/function-mesh/pull/254))
```

> - [controller] fix package download path not same as the executing command path ([#256](https://github.com/streamnative/function-mesh/pull/256))

Reviewer: ```suggestion
- [Controller] Fix the issue that the package download path is different from the executing command path ([#256](https://github.com/streamnative/function-mesh/pull/256))
```

> - [controller] support metrics based HPA by upgrade apis to k8s.io/api/autoscaling/v2beta2 ([#245](https://github.com/streamnative/function-mesh/pull/245))

Reviewer: ```suggestion
- [Controller] Support metrics-based HPA by upgrade APIs to `k8s.io/api/autoscaling/v2beta2` ([#245](https://github.com/streamnative/function-mesh/pull/245))
```

> - [controller] fix misuse of kubernetes namespace in controller ([#259](https://github.com/streamnative/function-mesh/pull/259))

Reviewer: ```suggestion
- [Controller] Fix misuse of Kubernetes namespace in Controller ([#259](https://github.com/streamnative/function-mesh/pull/259))
```

> - [controller] support download package with auth/tls ([#257](https://github.com/streamnative/function-mesh/pull/257))

Reviewer: ```suggestion
- [Controller] Support downloading the package with authentication or TLS ([#257](https://github.com/streamnative/function-mesh/pull/257))
```

> - [mesh-worker-service] support create function with package URL ([#261](https://github.com/streamnative/function-mesh/pull/261))

Reviewer: ```suggestion
- [Function Mesh Worker service] Support creating a function with the package URL ([#261](https://github.com/streamnative/function-mesh/pull/261))
```

> - [tests] add integration tests for pulsar's package management service ([#268](https://github.com/streamnative/function-mesh/pull/268))

Reviewer: ```suggestion
- [Test] Add integration tests for Pulsar package management service ([#268](https://github.com/streamnative/function-mesh/pull/268))
```

> - [code-base] remove vendor folder in codebase ([#267](https://github.com/streamnative/function-mesh/pull/267))

Reviewer: ```suggestion
- [Codebase] Remove the vendor folder in the codebase ([#267](https://github.com/streamnative/function-mesh/pull/267))
```
